tests/src/test_firecommand.cpp

```diff
@@ -0,0 +1,145 @@
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+*/
+
+#include "es-test.hpp"
+
+// Include only the tested class's header.
+#include "../../source/FireCommand.h"
+
+// ... and any system includes needed for the test file.
+
+namespace { // test namespace
+
+// #region mock data
+// #endregion mock data
+
+
+
+// #region unit tests
+SCENARIO( "Creating a FireCommand instance", "[firecommand]") {
+	GIVEN( "an instance" ) {
+		FireCommand command;
+		THEN( "it has the correct default properties" ) {
+			CHECK_FALSE( command.IsFiring() );
+			CHECK_FALSE( command.HasFire(0) );
+			CHECK_FALSE( command.Aim(0) );
+		}
+	}
+}
+
+SCENARIO( "A FireCommand instance is being copied", "[firecommand]" ) {
+	FireCommand command;
+	GIVEN( "a specific bitset" ) {
+		command.SetHardpoints(5);
+		command.SetFire(0);
+		command.SetFire(3);
+		command.SetAim(2, 1.);
+
+		WHEN( "the copy is made" ) {
+			auto copy = command;
+			THEN( "the copy has the correct properties" ) {
+				CHECK( copy.IsFiring() == command.IsFiring() );
+				CHECK( copy.HasFire(0) == command.HasFire(0) );
+				CHECK( copy.HasFire(1) == command.HasFire(1) );
+				CHECK( copy.HasFire(2) == command.HasFire(2) );
+				CHECK( copy.HasFire(3) == command.HasFire(3) );
+				CHECK( copy.HasFire(4) == command.HasFire(4) );
+				CHECK( copy.Aim(0) == Approx(command.Aim(0)) );
+				CHECK( copy.Aim(1) == Approx(command.Aim(1)) );
+				CHECK( copy.Aim(2) == Approx(command.Aim(2)) );
+				CHECK( copy.Aim(3) == Approx(command.Aim(3)) );
+				CHECK( copy.Aim(4) == Approx(command.Aim(4)) );
+			}
+			THEN( "the two bitsets are independent" ) {
+				command.SetAim(1, -1.);
+				CHECK( command.Aim(1) == Approx(-1) );
+				CHECK_FALSE( copy.Aim(0) );
+
+				copy.SetFire(4);
+				CHECK_FALSE( command.HasFire(4) );
+				CHECK( copy.HasFire(4) );
+			}
+		}
+	}
+}
+
+SCENARIO( "A FireCommand instance is being used", "[firecommand]") {
+	GIVEN( "an empty FireCommand" ) {
+		FireCommand command;
+		THEN( "resizing it works" ) {
+			command.SetHardpoints(20);
+			command.SetFire(18);
+			CHECK( command.HasFire(18) );
+		}
+	}
+	GIVEN( "a FireCommand of a specific size" ) {
+		FireCommand command;
+		command.SetHardpoints(10);
+
+		THEN( "setting and testing weapon works" ) {
+			CHECK_FALSE( command.IsFiring() );
+
+			command.SetAim(4, -1.);
+			CHECK_FALSE( command.HasFire(4) );
+			CHECK( command.Aim(4) == Approx(-1.) );
+
+			CHECK_FALSE( command.HasFire(5) );
+			command.SetFire(5);
+			CHECK( command.HasFire(5) );
+
+			CHECK( command.IsFiring() );
+		}
+		THEN( "clearing it works" ) {
+			command.SetFire(3);
+			command.SetAim(2, 1.);
+			CHECK( command.HasFire(3) );
+			CHECK( command.Aim(2) == Approx(1.) );
+			CHECK( command.IsFiring() );
+
+			command.Clear();
+			CHECK_FALSE( command.HasFire(3) );
+			CHECK_FALSE( command.Aim(2) == Approx(1.) );
+			CHECK_FALSE( command.IsFiring() );
+		}
```

Comment:
Split these into sections that focus just on setting & reading fire, setting & reading aim (and make sure both work with the first index).

The "clear" bits should be in a predicated WHEN, e.g.
```
Given a FireCommand of a specific size
  (set up the fire command)
  and_given an index is firing
    (call setfire)
    When clear is called
      (call clear)
      Then ...
        (assertions for this "story")
```
a failed `REQUIRE` in one of the parent blcoks will prevent entering the leaf blocks.